starlette/testclient.py

```diff
@@ -22,58 +26,63 @@ class _MockOriginalResponse(object):
     it was made using urllib3.
     """
 
-    def __init__(self, headers):
+    def __init__(self, headers: BytesPairs) -> None:
         self.msg = _HeaderDict(headers)
         self.closed = False
 
-    def isclosed(self):
+    def isclosed(self) -> bool:
         return self.closed
 
 
 class _Upgrade(Exception):
-    def __init__(self, session):
+    def __init__(self, session: "WebSocketTestSession") -> None:
         self.session = session
 
 
-def _get_reason_phrase(status_code):
+def _get_reason_phrase(status_code: int) -> str:
     try:
         return http.HTTPStatus(status_code).phrase
     except ValueError:
         return ""
 
 
 class _ASGIAdapter(requests.adapters.HTTPAdapter):
-    def __init__(self, app: typing.Callable, raise_server_exceptions=True) -> None:
+    def __init__(self, app: ASGIApp, raise_server_exceptions: bool = True) -> None:
         self.app = app
         self.raise_server_exceptions = raise_server_exceptions
 
-    def send(self, request, *args, **kwargs):
-        scheme, netloc, path, params, query, fragement = urlparse(request.url)
+    def send(  # type: ignore
+        self, request: requests.PreparedRequest, *args: typing.Any, **kwargs: typing.Any
```

Comment:
Interesting that this is marked `type: ignore` - are we not adhering to the same interface as `requests.adapters.HTTPAdapter` here?